test/specs/repo/queryservice.ts

```diff
@@ -151,4 +152,75 @@ describe( 'QueryService', () => {
 		// timestamp always shows
 		assert( resultText.includes( 'wikibase:timestamp' ) );
 	} );
+
+	it( 'Should show results for a select query', async () => {
+		await QueryServiceUIPage.open( 'SELECT * where { ?a ?b ?c }' );
+		await QueryServiceUIPage.submit();
+		expect(
+			( await QueryServiceUIPage.resultTable.$( 'tbody' ).$$( 'tr' ) ).length
+		).toBeGreaterThan( 0 );
+	} );
+
+	it( 'Should show list of properties', async () => {
+		await QueryServiceUIPage.open( `SELECT ?property ?propertyType ?propertyLabel ?propertyDescription ?propertyAltLabel WHERE {
+			?property wikibase:propertyType ?propertyType .
+			SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],en". }
+		  }
+		  ORDER BY ASC(xsd:integer(STRAFTER(STR(?property), 'P')))` );
+		await QueryServiceUIPage.submit();
+		await expect(
+			QueryServiceUIPage.resultTable.$( 'th[data-field="property"]' )
+		).toExist();
+		await expect(
+			QueryServiceUIPage.resultTable.$( 'th[data-field="propertyType"]' )
+		).toExist();
+		await expect(
+			QueryServiceUIPage.resultTable.$( 'th[data-field="propertyLabel"]' )
+		).toExist();
+		await expect(
+			QueryServiceUIPage.resultTable.$( 'th[data-field="propertyDescription"]' )
+		).toExist();
+		await expect(
+			QueryServiceUIPage.resultTable.$( 'th[data-field="propertyAltLabel"]' )
+		).toExist();
+		expect(
+			( await QueryServiceUIPage.resultTable.$( 'tbody' ).$$( 'tr' ) ).length
+		).toBeGreaterThan( 0 );
+	} );
+
+	it( 'Should show a property connected to item', async () => {
+		const propertyId = await WikibaseApi.createProperty(
+			wikibasePropertyString.urlName
+		);
+		const data = {
+			claims: [
+				{
+					mainsnak: {
+						snaktype: 'value',
+						property: propertyId,
+						datavalue: {
+							value: 'test-property',
+							type: wikibasePropertyString.urlName
+						}
+					},
+					type: 'statement',
+					rank: 'normal'
+				}
+			]
+		};
+
+		const itemId = await WikibaseApi.createItem(
+			getTestString( 'test-item-label' ),
+			data
+		);
+
+		await QueryServiceUIPage.open( `SELECT (COUNT(*) AS ?count)
+		WHERE {
+		  ?item <${testEnv.vars.WIKIBASE_HOST}/prop/direct/${propertyId}> <${testEnv.vars.WIKIBASE_HOST}/entity/${itemId}> .
+		}` );
+
+		expect(
+			( await QueryServiceUIPage.resultTable.$( 'tbody' ).$$( 'tr' ) ).length
+		).toBeGreaterThan( 0 );
```

Comment:
Maybe we can check a bit more here? Like the actual values in the table?

Reply:
Updated!